test/components/collision_detection_make_activity_test.dart

```diff
@@ -0,0 +1,225 @@
+import 'dart:math';
+import 'package:flame/components.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/geometry.dart';
+import 'package:flame/src/geometry/collision_detection.dart'
+    as CollisionDetectionFunction;
+import 'package:test/test.dart';
+
+class _TestBlock extends PositionComponent with Hitbox, Collidable {
+  Set<Collidable> oncollidables = {};
+  Set<Collidable> cachecollidables = {};
+  _TestBlock() {
+    addShape(HitboxCircle());
+  }
+  @override
+  void onCollision(Set<Vector2> points, Collidable other) {
+    oncollidables.add(other);
+    super.onCollision(points, other);
+  }
+}
+
+class _TestBlockA extends _TestBlock {
+  _TestBlockA(Vector2 pos, Vector2 size) {
+    activeCollidable = false;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockB extends _TestBlock {
+  _TestBlockB(Vector2 pos, Vector2 size) {
+    activeCollidable = false;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockC extends _TestBlock {
+  _TestBlockC(Vector2 pos, Vector2 size) {
+    activeCollidable = true;
+    position = pos;
+    this.size = size;
+  }
+}
+
+class _TestBlockD extends _TestBlock {
```

Comment:
you dont need to create multiple classes if they are all the same, just create multiple instances of the same class